linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_sysadmin_actions/ansible/shared.yml

```diff
@@ -0,0 +1,43 @@
+# platform = multi_platform_rhel
+# reboot = false
+# complexity = low
+
+- name: Search /etc/audit/rules.d for other sudoers audit rules
+  find:
+    paths: /etc/audit/rules.d
+    recurse: false
+    contains: -k sudoers$
+    patterns: '*.rules'
+  register: find_sudoers
+  tags:
+    @ANSIBLE_TAGS@
+- name: If existing sudoers ruleset not found, use /etc/audit/rules.d/privileged.rules as the recipient for the rule
+  set_fact:
+    all_files:
+    - /etc/audit/rules.d/privileged.rules
+  when: find_sudoers.matched == 0
+- name: Use matched file as the recipient for the rule
+  set_fact:
+    all_files:
+    - '{{ find_sudoers.files | map(attribute=''path'') | list | first }}'
+  when: find_sudoers.matched > 0
+- name: Inserts/replaces the sudoers rule in rules.d
+  lineinfile:
```

Comment:
Add `regexp` parameter to `lineinfile` section, so it will overwrite existing sudoers with correct access bits (`-p`).